File: app/src/main/java/com/atls/aeterna/MainActivity.java
package com.atls.aeterna;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.TargetApi;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.provider.Settings;
import android.util.Log;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "interaction";
    public static int OVERLAY_PERMISSION_REQ_CODE = 1;

    @TargetApi(Build.VERSION_CODES.M)
    public void checkPermissionOverlay() {
        if (!Settings.canDrawOverlays(this)) {
            Toast.makeText(MainActivity.this, "Overlay permission required!", Toast.LENGTH_LONG).show();
            Intent intentSettings = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION);
            startActivityForResult(intentSettings, OVERLAY_PERMISSION_REQ_CODE);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG, "onCreate");
        setContentView(R.layout.activity_main);
        checkPermissionOverlay();
        Intent svc = new Intent(this, OverlayService.class);
        startService(svc);
        finish();
    }
}

File: app/src/main/java/team/atls/aeterna/MainActivity.kt
package team.atls.aeterna

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        try {
            System.loadLibrary("interaction")
        } catch (e: UnsatisfiedLinkError) {
            Log.e("Interaction", "Load libary ERROR: $e")
            return
        }

        val username = findViewById<EditText>(R.id.username);
        val password = findViewById<EditText>(R.id.password);
        val event = findViewById<TextView>(R.id.event);
        val generate: Button = findViewById(R.id.generate);

        generate.setOnClickListener(object: View.OnClickListener {
            override fun onClick(v: View?) {
                event.text = PokerBros.login(username.text.toString(), password.text.toString());
            }
        })
    }
}